lib/crowbar/client/command/role.rb

```diff
@@ -65,16 +80,19 @@ module Role
                 Request.instance.role_show(barclamp, role) do |request|
                   case request.code
                   when 200
-                    body = begin
-                      JSON.parse(request.body)
-                    rescue
-                      []
-                    end
+                    formatter = Formatter::Array.new(
```

Comment:
And in general that's not all I want to refactor and make more abstract, just the beginning :)
I will add some tests as well when I get back to this.